Gruntfile.js

```diff
@@ -26,8 +26,16 @@ module.exports = function(grunt) {
     concat: {
       options: {
         banner: '<%= meta.banner %>' + '\n' +
-          '(function ( window, angular, undefined ) {' + '\n',
-        footer: '})( window, window.angular );'
+        "(function (factory) {\n" +
+        "  if (typeof define === 'function' && define.amd) {\n" +
+        "    define(['exports', 'angular'], factory);\n" +
+        "  } else if (typeof module === 'object' && typeof module.exports === 'object') {\n" +
+        "    factory(module.exports, require('angular'));\n" +
+        "  } else if (root.hasOwnProperty('angular')) {\n" +
+        "    factory((root.exports || (root.exports = {})), root.angular);\n" +
```

Comment:
Actually I don't understand why the module should be attached to `root.exports` object, this may cause conflicts like #288